apps/openmw/mwworld/store.cpp

```diff
@@ -179,16 +185,19 @@ namespace MWWorld
         return ptr;
     }
     template<typename T>
-    void Store<T>::load(ESM::ESMReader &esm, const std::string &id) 
+    RecordId Store<T>::load(ESM::ESMReader &esm) 
     {
-        std::string idLower = Misc::StringUtils::lowerCase(id);
+        T record;
+        bool isDeleted = false;
 
-        std::pair<typename Static::iterator, bool> inserted = mStatic.insert(std::make_pair(idLower, T()));
+        record.load(esm, isDeleted);
+        Misc::StringUtils::toLower(record.mId);
+
+        std::pair<typename Static::iterator, bool> inserted = mStatic.insert(std::make_pair(record.mId, record));
         if (inserted.second)
             mShared.push_back(&inserted.first->second);
 
-        inserted.first->second.mId = idLower;
-        inserted.first->second.load(esm);
```

Comment:
To be able to merge records we can split the load method into two methods: loadId() and loadData(). But in this case, both methods must have a deleted flag as a parameter. To know whether a record is deleted we must checked them both. 

What's your opinions about this, @zinnschlag, @scrawl?